libs/dh/wholesale/feature-missing-measurements-log/src/lib/request-log.ts

```diff
@@ -0,0 +1,188 @@
+          [label]="t('period')"
+          [range]="true"
+          [min]="minDate"
+          [max]="maxDate"
+          [formControl]="form.controls.period"
+          data-testid="missingMeasurementsLog.requestLog.datePeriod"
+        >
+          @if (form.controls.period.errors?.['maxDays']) {
+            <watt-field-error>{{ t('maxPeriodLength') }}</watt-field-error>
+          } @else if (form.controls.period.errors?.['monthOnly']) {
+            <watt-field-error>{{ t('monthOnlyError') }}</watt-field-error>
+          }
+        </watt-datepicker>
+        <dh-calculations-grid-areas-dropdown
+          [period]="period()"
+          [control]="form.controls.gridAreaCodes"
+          [multiple]="true"
+          [preselect]="false"
+        />
+      </form>
+      <watt-modal-actions>
+        <watt-button variant="secondary" (click)="modal.close(false)">
+          {{ t('cancel') }}
+        </watt-button>
+        <watt-button variant="primary" formId="request-log" type="submit">
+          {{ t('request') }}
+        </watt-button>
+      </watt-modal-actions>
+    </watt-modal>
+  `,
+})
+export class DhWholesaleMissingMeasurementsLogRequestLog {
+  form = new FormGroup({
+    period: dhMakeFormControl<WattRange<string>>(null, [
+      Validators.required,
+      WattRangeValidators.required,
+      WattRangeValidators.maxDays(31),
+    ]),
+    gridAreaCodes: dhMakeFormControl<string[]>([], Validators.required),
+  });
+  period = toSignal(
+    this.form.controls.period.valueChanges.pipe(
+      map((interval) => {
+        return interval
+          ? {
+              interval: {
+                start: dayjs(interval.start).toDate(),
+                end: dayjs(interval.end).toDate(),
+              },
+            }
+          : undefined;
+      })
+    )
+  );
+  modal = viewChild(WattModalComponent);
+  open = () => this.modal()?.open();
+  close = (result: boolean) => this.modal()?.close(result);
+
+  minDate = getMinDate();
+  maxDate = getMaxDate();
+
+  // Request mutation handling
+  request = mutation(RequestMissingMeasurementsLogDocument);
+  toast = injectToast();
+  toastEffect = effect(() => this.toast(this.request.status()));
+  handleSubmit = () => {
+    if (!this.form.valid) return;
+    this.close(true);
+
+    this.request.mutate({
+      variables: { input: this.makeRequestMissingMeasurementsLogInput() },
+    });
+  };
+
+  makeRequestMissingMeasurementsLogInput = (): RequestMissingMeasurementsLogInput => {
+    const { gridAreaCodes, period } = this.form.value;
+
+    // Satisfy the type checker, since fields should be defined at this point (due to validators)
+    assertIsDefined(period);
+
+    return {
+      period: {
+        start: dayjs(period.start).toDate(),
+        end: dayjs(period.end).toDate(),
+      },
+      gridAreaCodes: gridAreaCodes ?? [],
+    };
+  };
+
+  constructor() {
+    effect(() => {
+      const modal = this.modal();
+      setTimeout(() => {
+        modal?.open();
```

Comment:
Question: Is this temporarily?

Reply:
Yes it is temporary, currently we don't show anything on the page besides the modal for requesting missing measurements. As I understand we will later show something in the table behind it. Until then we just open up the modal automatically to spare a click for the user.

Reply:
This has now been removed due to the refactor to a `routed` modal.